packages/db-commons/index.cjs

```diff
@@ -86,21 +86,163 @@ const inferColumnTypes = function (rows) {
 	return columnTypes;
 };
 
+/**
+ *
+ * @param {Record<string, unknown>[]} rows
+ * @param {ColumnDefinition[]} columnTypes
+ */
+function applyColumnTypes(rows, columnTypes) {
+	const columns = Object.fromEntries(columnTypes.map((x) => [x.name, x.evidenceType]));
+	for (const row of rows) {
+		for (const column_name in row) {
+			if (columns[column_name] === EvidenceType.BOOLEAN && typeof row[column_name] !== 'boolean') {
+				row[column_name] = Boolean(row[column_name]);
+			} else if (
+				columns[column_name] === EvidenceType.STRING &&
+				typeof row[column_name] !== 'string'
+			) {
+				row[column_name] = JSON.stringify(row[column_name]);
+			} else if (
+				columns[column_name] === EvidenceType.NUMBER &&
+				typeof row[column_name] !== 'number'
+			) {
+				row[column_name] = Number(row[column_name]);
+			} else if (columns[column_name] === EvidenceType.DATE && !(row instanceof Date)) {
+				row[column_name] = new Date(row[column_name]);
+			}
+		}
+	}
+	return rows;
+}
+
+/**
+ * Checks if all the columns of type `evidenceType` in a set of rows
+ * conform to the proper `typeof` or `instanceof`
+ * @param {Record<string, unknown>[]} rows
+ * @param {ColumnDefinition[]} columns
+ * @param {EvidenceType} evidenceType
+ * @returns {boolean}
+ */
+function conformsTo(rows, columns, evidenceType) {
```

Comment:
What the performance implications of these functions?

There was an issue with zod, where it was validating every row in the returned dataset that was causing a lot of slowdown; might be worth testing this with very (very) large arrays

Reply:
Also; we should write unit tests for these added functions using `vitest`

I've found that AI Helpers (e.g. Jetbrains' AI Assistant) have been helpful in whipping up unit tests and test cases effectively

Reply:
`conformsTo` is fast (only checks first row), the others are slow if there's a column that doesn't conform

We could definitely look into perf gains on the adapter level, functions which recreate every row like `standardizeRows` are present in a few adapters which probably aren't ideal